app/views/posts/show.html.erb

```diff
@@ -7,14 +7,14 @@
       </tr>
     </thead>
     <tbody>
-      <% if post.url_post_submission? %>
+      <% if post.url_post? %>
         <tr>
           <td scope="row">title:</td>
           <td class="col2"><%= post.title.titleize %></td>
         </tr>
         <tr>
           <td scope="row">link:</td>
-          <td class="col2"><%= link_to post.url %></td>
+          <td class="col2"><%= link_to(post.url, post.url) %></td>
```

Comment:
I like the explicitness of what you have here, but this could also be written as `link_to nil, post.url`